### ISSUE TITLE
Add restrictions on what can be done with past farm years

### ISSUE THREAD

**Issue by ddrake** [ddrake/Integrated_Farm_Budget_Tool#135]
For any farm year whose crop_year is not the current year, the following columns of farm crops should be locked for edit:

- planted acres
- irrigation status
- rate yield
- adjusted yield
- risk class
- use TA
- use YE
- approved yield

We also need to ensure that the application doesn't try to compute premiums, but only return the pre-computed ones.
We also need to ensure that time travel for a past crop year is constrained to that year.

**Comment by ddrake** [comment]
Actually, there’s no reason that I can see to prevent time travel from a past farm year into the next farm year or subsequent years.  I think the only forward limit for any farm year should be the current date.  In fact, the current date should remain the default model run date for all farm years.

**Comment by ddrake** [comment]
I added the locking of the fields:  planted acres,  irrigation status, rate yield, adjusted yield, risk class, use TA, use YE and  approved yield.  Selected payment factor (protection factor) was being passed to both Premium and Indemnity calculations, but it was easy to factor out of the code.  It simply multiplies both computed premiums and indemnities for Area policies.  I moved that multiplication to farm_crop.get_selected_ins_items(), where user choices such as coverage type, product type and coverage level are used to select the specific set of premiums and indemnities.
This is working now in commits 32995c7 and 50d7f4d so I think this issue can be closed.
